superagi/controllers/knowledge.py

```diff
@@ -0,0 +1,180 @@
+from fastapi_sqlalchemy import db
+from fastapi import HTTPException, Depends, Query, status
+from fastapi_jwt_auth import AuthJWT
+from fastapi import APIRouter
+from superagi.config.config import get_config
+from superagi.helper.auth import get_user_organisation
+from superagi.models.knowledge import Knowledge
+from superagi.models.knowledge_config import KnowledgeConfig
+from superagi.models.vector_db_index_collection import VectorIndexCollection
+from superagi.models.vector_db import Vectordb
+from superagi.models.marketplace_stats import MarketPlaceStats
+from superagi.llms.openai import OpenAi
+from superagi.helper.pinecone_helper import PineconeHelper
+from superagi.helper.qdrant_helper import QdrantHelper
+from superagi.helper.knowledge_helper import KnowledgeHelper
+from superagi.models.vector_db_config import VectordbConfig
+from superagi.helper.auth import get_user_organisation
+
+router = APIRouter()
+
+
+@router.get("/get/list")
+def handle_marketplace_operations_list(
+        page: int = Query(None, title="Page Number"),
+        organisation = Depends(get_user_organisation)
+):
+    """
+    Handle marketplace operation list.
+
+    Args:
+        page (int, optional): The page number for pagination. Defaults to None.
+
+    Returns:
+        dict: The response containing the marketplace list.
+
+    """
+
+    marketplace_organisation_id = int(get_config("MARKETPLACE_ORGANISATION_ID"))
+    marketplace_knowledges = Knowledge.fetch_marketplace_list(db.session, page, marketplace_organisation_id)
+    marketplace_knowledges_with_install = Knowledge.get_knowledge_installed_details(db.session, marketplace_knowledges,
+                                                                              organisation)
+    for knowledge in marketplace_knowledges_with_install:
+        knowledge["install_number"] = MarketPlaceStats.get_knowledge_installation_number(db.session, knowledge["id"])
+    return marketplace_knowledges_with_install
+
+@router.get("/user/list")
+def get_user_knowledge_list(Authorize: AuthJWT = Depends(), organisation = Depends(get_user_organisation)):
+    organisation_id = organisation.id
+    marketplace_organisation_id = int(get_config("MARKETPLACE_ORGANISATION_ID"))
```

Comment:
logic does not seem right. local one cant even fetch MARKETPLACE_ORGANISATION_ID. In local nobody sets MARKETPLACE_ORGANISATION_ID